### PR TITLE
[SYCL][ESIMD] Use intrinsic for named_barrier_signal

### PR DESCRIPTION
**Pull request by sarnex**
We don't want to use raw_send here. I manually tested this on PVC with the required driver version.

### CODE REVIEW

File: llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp [hunk 1]
```diff
@@ -494,6 +494,7 @@ class ESIMDIntrinDescTable {
         {"dpasw", {"dpasw", {a(0), a(1), a(2), t(0)}}},
         {"dpasw_nosrc0", {"dpasw.nosrc0", {a(0), a(1), t(0)}}},
         {"nbarrier", {"nbarrier", {a(0), a(1), a(2)}}},
+        {"nbarrier_arrive", {"nbarrier.arrive", {a(0), a(1), a(2), a(3)}}},
         {"raw_send_nbarrier_signal",
```
Comment (v-klochkov) [inline llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp:498]: Can you please also remove this line with 'raw_send_nbarrier_signal'

**Review by v-klochkov — COMMENTED** [review]
Looks good to me. 
I would also remove the test 'llvm/test/SYCLLowerIR/esimd_lower_nbarriers.ll'

File: llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp (hunk 1 above)
Comment (sarnex) [reply llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp]: fixed in https://github.com/intel/llvm/pull/12982/commits/7fa62a1e0a2b8d81b4fcc4c6f45861820cbfeb7b, thanks

**Comment by sarnex** [comment]
@v-klochkov Both comments should be addressed in https://github.com/intel/llvm/pull/12982/commits/7fa62a1e0a2b8d81b4fcc4c6f45861820cbfeb7b, thanks

**Review by v-klochkov — APPROVED** [review]

**Comment by sarnex** [comment]
@intel/dpcpp-tools-reviewers Can someone please review? Just updating the VC intrinsics commit hash. Thanks

**Comment by sarnex** [comment]
Ping @intel/dpcpp-tools-reviewers, should be a trivial review

**Review by LU-JOHN — APPROVED** [review]
LGTM

**Comment by sarnex** [comment]
```
Failed Tests (1)
  SYCL :: Graph/Explicit/host_task_multiple_deps.cpp
```
not related

**Pull request merged**